ui-admin/src/api/enrolleeSearch.ts

```diff
@@ -0,0 +1,191 @@
+  label: string, value: string
+}
+
+export type StringFacet = BaseFacet & {
+  type: 'STRING',
+  options: FacetOption[]
+}
+
+export type StableIdStringArrayFacet = BaseFacet & {
+  type: 'STABLEID_STRING',
+  options: FacetOption[]
+  stableIdOptions: FacetOption[]
+}
+
+export type Facet = StringFacet | StableIdStringArrayFacet | IntRangeFacet
+
+export const SAMPLE_FACETS: Facet[] = [{
+  category: 'profile',
+  keyName: 'age',
+  label: 'Age',
+  type: 'INT_RANGE',
+  max: 150,
+  min: 0
+}, {
+  category: 'profile',
+  keyName: 'sexAtBirth',
+  label: 'Sex at birth',
+  type: 'STRING',
+  options: [
+    { value: 'male', label: 'male' },
+    { value: 'female', label: 'female' },
+    { value: 'other', label: 'other' },
+    { value: 'unknown', label: 'unknown' }
+  ]
+}, {
+  category: 'participantTask',
+  keyName: 'status',
+  label: 'Task status',
+  type: 'STABLEID_STRING',
+  options: [
+    { value: 'COMPLETE', label: 'Complete' },
+    { value: 'IN_PROGRESS', label: 'In progress' },
+    { value: 'NEW', label: 'New' }
+  ],
+  stableIdOptions: [
+    { value: 'oh_oh_consent', label: 'Consent' },
+    { value: 'oh_oh_basicInfo', label: 'Basics' },
+    { value: 'oh_oh_cardioHx', label: 'Cardio History' }
+  ]
+}]
+
+
+/**
+ *
+ */
+export const newFacetValue = (facet: Facet, facetValue?: object): FacetValue => {
+  if (facet.type === 'INT_RANGE') {
+    return new IntRangeFacetValue(facet, facetValue as IntRangeFacetValueFields)
+  } else if (facet.type === 'STABLEID_STRING') {
+    const newValues = facetValue ? (facetValue as StableIdStringArrayFacetValueFields).values.map(stableIdVal =>
+      new StableIdStringValue(stableIdVal.stableId, stableIdVal.values)
+    ) : []
+    return new StableIdStringArrayFacetValue(facet, { values: newValues })
+  }
+  return new StringFacetValue(facet, facetValue as StringFacetValueFields)
+}
+
+/**
+ *
+ */
+export const facetValuesToString = (facetValues: FacetValue[]): string => {
+  const paramObj: Record<string, Record<string, object>> = {}
+  facetValues
+    .filter(facetValue => !facetValue.isDefault())
+    .forEach(facetValue => {
+      const category = facetValue.facet.category
+      const keyName = facetValue.facet.keyName
+      // strip out the 'facet' property since we don't need to send that
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { facet: _, ...values } = facetValue
+      paramObj[category] = paramObj[category] ?? {}
+      paramObj[category][keyName] = values
+    })
+  return JSON.stringify(paramObj)
+}
+
+/**
+ *
+ */
+export const facetValuesFromString = (paramString: string, facets: Facet[]): FacetValue[] => {
+  const facetValues = []
+  const paramObj = JSON.parse(paramString)
+
+  for (const categoryName in paramObj) {
+    const category = paramObj[categoryName]
+    for (const keyName in category) {
+      const matchedFacet = facets.find(facet => facet.category === categoryName && facet.keyName === keyName) as Facet
```

Comment:
Is there guaranteed to be a matching facet?

Reply:
no, good point, will add code to ignore mismatched facets